app/lib/search/mem_index.dart

```diff
@@ -480,15 +480,15 @@ class PackageNameIndex {
     final singularWord = word.length <= 3 || !word.endsWith('s')
         ? word
         : word.substring(0, word.length - 1);
+    final parts = singularWord.length <= 3
```

Comment:
nit: Should this have been lazy (late)? we only use it if we get past the first condition inside the loop.

Reply:
I'd think that most cases will trigger it anyway, but it wouldn't hurt to have that calculated only on-demand.